web_poet/requests.py

```diff
@@ -82,18 +84,56 @@ class HttpClient:
     def __init__(self, request_downloader: Callable = None):
         self._request_downloader = request_downloader or _perform_request
 
+    @staticmethod
+    def _handle_status(
+        response: HttpResponse,
+        request: HttpRequest,
+        *,
+        allow_status: List[_Status] = None,
+    ) -> None:
+        allow_status_normalized = list(map(str, as_list(allow_status)))
+        allow_all_status = any(
+            [True for s in allow_status_normalized if "*" == s.strip()]
+        )
+
+        if (
+            allow_all_status
+            or response.status is None  # allows serialized responses from tests
+            or response.status < 400
+            or str(response.status) in allow_status_normalized
+        ):
+            return
+
+        if 400 <= response.status < 500:
+            kind = "Client"
+        elif 500 <= response.status < 600:
+            kind = "Server"
```

Comment:
If that's desirable, you can get error description from https://docs.python.org/3/library/http.html, instead of having a logic here

Reply:
A good suggestion! Used this on https://github.com/scrapinghub/web-poet/pull/38/commits/1401a514106514d2910bcb6dc15d4d5909d5b453.